### DISCUSSION TITLE
Руководство ПЛИ, п.3.3 автор документа

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#271] [Лабораторная РГ]
3.3 Автор <author>
Элемент author описывает создателя документа и содержит данные о человеке, который составил документ "Протокол лабораторного исследования" (создал в информационной системе). Автор документа представляет лицо, которое на основе своих знаний и навыков предоставил информацию для документа. Если автор документа получает и обрабатывает информацию от другого участника, фильтрует её, алгоритмически преобразует и получает на её основе новую информацию, то этот другой участник тоже должен быть указан как автор.

**Comment by eugenekogan-spb** [comment]
3.3.0. элемент author
У1-13:Элемент ClinicalDocument ОБЯЗАН иметь один [1..1] элемент author, содержащий сведения об авторе документа, который:

**Reply by eugenekogan-spb** [reply]
Кажется, требование иметь не более чем один элемент author противоречит приведенному выше требованию "...то этот другой участник тоже должен быть указан как автор."

**Reply by eugenekogan-spb** [reply]
В Composition необходимо указать автора документа
```
* author = Reference(practitionerrole-lab-doctor-helix-altay)
```

**Comment by eugenekogan-spb** [comment]
3.3.1. time
1. ДОЛЖЕН иметь один [1..1] элемент time (атрибут @nullFlavor должен иметь значение "NI");

**Reply by eugenekogan-spb** [reply]
Спецификация не уточняет, чем это время должно отличаться от effectiveDate, описанного [в заголовке документа](https://github.com/fhir-ru/core/discussions/268#discussioncomment-10432021), поэтому предлагаю принять их одинаковыми.
```
* date = "2013-02-01T12:30:02Z"
```

**Reply by OlegPenzin** [reply]
насколько я помню существует какая-то особенность сверки с датой наложения ЭЦП на СЭМД, которая как раз требует, чтобы эту дату вычитывали из подписи (которая станет не валидной, если заменить содержимое в XML после подписания)

**Reply by eugenekogan-spb** [reply]
Встреча 24.12.24: эту дату в СЭМД просто не заполняем.

**Comment by eugenekogan-spb** [comment]
3.3.2. assignedAuthor

2. ОБЯЗАН иметь один [1..1] элемент assignedAuthor, который:

**Comment by eugenekogan-spb** [comment]
3.3.2.1 id

2.1. ОБЯЗАН иметь один [1..1] элемент id (атрибуты @root, @extension должны быть представлены с не пустыми значениями; атрибут @root должен быть сформирован по правилу формирования идентификаторов персонала, т.е. "OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.70", атрибут @extension должен соответствовать идентификатору персонала), содержащий уникальный идентификатор автора в МИС;

**Reply by eugenekogan-spb** [reply]
```
Instance: practitionerrole-lab-doctor-helix-altay
InstanceOf: PractitionerRole
Usage: #inline

//identifier[misPractitionerRoleId] (OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.70)
* identifier[+]
  * value = "423"
  * system = "1.2.643.5.1.13.13.12.2.22.27143.100.1.1.70" 
  * type = http://terminology.hl7.org/CodeSystem/v2-0203#ESN
```

**Comment by eugenekogan-spb** [comment]
3.3.2.2. id
2.2. ОБЯЗАН иметь один [1..1] элемент id (атрибут @root должен иметь значение "1.2.643.100.3", атрибут @extension должен соответствовать номеру СНИЛС), содержащий СНИЛС автора;

**Reply by eugenekogan-spb** [reply]
```
Instance: practitionerrole-lab-doctor-helix-altay
InstanceOf: PractitionerRole
Usage: #inline

* practitioner = Reference(practitioner-lab-doctor-helix-altay)

Instance: practitioner-lab-doctor-helix-altay
InstanceOf: Practitioner
Usage: #inline

//* identifier[SNILS] 
* identifier[+]
  * value = "152-873-614 14"
  * system = "http://fhir.ru/core/systems/snils"
  * type = http://terminology.hl7.org/CodeSystem/v2-0203#SB
```

**Comment by eugenekogan-spb** [comment]
3.3.2.3. code
2.3. ОБЯЗАН иметь один [1..1] элемент code (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1002", атрибут @codeSystemName должен иметь значение "Должности медицинских и фармацевтических работников", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий код должности автора;

**Reply by eugenekogan-spb** [reply]
```
* code
  * text = "Врач - бактериолог"
  * coding
    * system = "1.2.643.5.1.13.13.11.1002"
    * code = #17
```

**Reply by OlegPenzin** [reply]
не хватает display и version

```
* code
  * text = "Врач - бактериолог"
  * coding
    * system = "1.2.643.5.1.13.13.11.1002"
    * code = #17
    * display  = "Врач - бактериолог"
    * version  = "7.6"
```

**Reply by eugenekogan-spb** [reply]
На встрече 24.12.24: принято. Добавлено во всех Coding. По предложению О.Пензина также удален атрибут text, там, где он дублирует атрибут display.

**Comment by eugenekogan-spb** [comment]
3.3.2.4. address
2.4.МОЖЕТ иметь один [0..1] элемент addr (атрибут @use может быть представлен с не пустым значением), содержащий адрес места работы автора документа, который:
2.4.1. ОБЯЗАН иметь один [1..1] элемент streetAddressLine, содержащий адрес (текстом), который должен иметь не пустое наполнение;
Руководство по реализации СЭМД "Протокол лабораторного исследования" в соответствии с третьим уровнем требований архитектуры клинических документов версии 2 (CDA 2.0 HL7 v.3)
2.4.2. ОБЯЗАН иметь один [1..1] элемент address:stateCode (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.206", атрибут @codeSystemName должен иметь значение "Субъекты Российской Федерации", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий код субъекта РФ;
2.4.3. ДОЛЖЕН иметь один [1..1] элемент postalCode, содержащий почтовый индекс, который должен иметь не пустое наполнение;
2.4.4.ДОЛЖЕН иметь один [1..1] элемент fias:Address, содержащий идентификатор адреса по ФИАС, который:
2.4.4.1.ОБЯЗАН иметь один [1..1] элемент fias:AOGUID, содержащий глобальный уникальный идентификатор адресного объекта, который должен иметь не пустое наполнение;
2.4.4.2.ДОЛЖЕН иметь один [1..1] элемент fias:HOUSEGUID, содержащий глобальный уникальный идентификатор дома, который должен иметь не пустое наполнение;

**Reply by eugenekogan-spb** [reply]
аналогично пациенту

**Reply by OlegPenzin** [reply]
опять выбор между https://hl7.org/fhir/practitioner-definitions.html#Practitioner.address и https://hl7.org/fhir/practitionerrole-definitions.html#PractitionerRole.contact - поскольку здесь должен быть рабочий адрес контакты, предлагаю хранить их в PractitionerRole.contact.address

```
* contact
  //3.3 <!-- [0..1] Адрес места работы автора документа -->
  * address
    //<!-- R [1..1] Адрес (текстом) -->
    * text = "Алтайский край, город Барнаул, улица Сергея Ускова, дом 33"
  
    //<!-- R [1..1] Код субъекта РФ -->
    //* extension[regionRF]
    * extension[+]
      * url = "regionRF"
      * valueCoding
        * code = #2
        * system = "1.2.643.5.1.13.13.99.2.20"
        * version = "6.3"         
        * display = "Алтайский край"
          
    //<!-- [1..1] Почтовый индекс -->
    * postalCode = "656067"
 ```

**Reply by eugenekogan-spb** [reply]
Принято на встрече 24.12.24.
Дополнительное замечание: поскольку адрес места работы сотрудника часто совпадает с адресом его организации, то при написании примеров есть смысл сделать экземпляр адреса и на него ссылаться.

```
* contact
  * address = address-aleyskaya-hospital
```

**Comment by eugenekogan-spb** [comment]
3.3.2.5. telecom
5. МОЖЕТ иметь произвольное количество [0..*] элементов telecom (атрибут @use может быть представлен с не пустым значением, атрибут @value должен быть представлен с не пустым значением), содержащих контакты автора;

**Reply by eugenekogan-spb** [reply]
аналогично пациенту

**Reply by OlegPenzin** [reply]
контакты есть и в https://hl7.org/fhir/practitionerrole-definitions.html#PractitionerRole.contact , и в https://hl7.org/fhir/practitioner-definitions.html#Practitioner.telecom - поскольку здесь должны быть рабочие контакты, предлагаю хранить их в PractitionerRole.contact.telecom[]

```
//3.3 контакты
* contact[0]
  * telecom[0]
    * system = #phone
    * value = "+7 960 700 5745"
    * use = #work
    * rank = 1
  * telecom[1]
    * system = #email
    * value = "Smirnova_AI@mail.ru"
    * use = #work
    * rank = 2
```

**Comment by eugenekogan-spb** [comment]
3.3.2.6. assignedPerson
6. ОБЯЗАН иметь один [1..1] элемент assignedPerson, который:
6.1. ОБЯЗАН иметь один [1..1] элемент name, содержащий ФИО автора,
который:
6.1.1. ОБЯЗАН иметь один [1..1] элемент family, содержащий фамилию, который должен иметь не пустое наполнение;
6.1.2. ОБЯЗАН иметь один [1..1] элемент given, содержащий имя, который должен иметь не пустое наполнение;
6.1.3. МОЖЕТ иметь один [0..1] элемент identity:Patronymic, содержащий отчество, который должен иметь не пустое наполнение;

**Reply by eugenekogan-spb** [reply]
аналогично пациенту

**Comment by eugenekogan-spb** [comment]
3.3.2.7. representedOrganization

7. МОЖЕТ иметь один [0..1] элемент representedOrganization (атрибут @classCode должен иметь значение "ORG"), содержащий сведения о медицинской организации, где работает автор, который:
7.1. ОБЯЗАН иметь один [1..1] элемент id (атрибут @root должен иметь значение по справочнику "Реестр медицинских организаций Российской Федерации" (OID: 1.2.643.5.1.13.13.11.1461), поле "OID", атрибут @extension может быть представлен со значением по справочнику "ФРМО. Справочник структурных подразделений" (OID: 1.2.643.5.1.13.13.99.2.114), поле "OID структурного подразделения"), содержащий уникальный идентификатор медицинской организации и (при наличии) уникальный идентификатор структурного подразделения;
7.2. ОБЯЗАН иметь один [1..1] элемент name, содержащий наименование медицинской организации или ФИО ИП, который должен иметь не пустое наполнение;
7.3. МОЖЕТ иметь произвольное количество [0..*] элементов telecom (атрибут @use может быть представлен с не пустым значением, атрибут @value должен быть представлен с не пустым значением), содержащих контакты медицинской организации или ИП;
7.4. МОЖЕТ иметь один [0..1] элемент addr (атрибут @use может быть представлен с не пустым значением), содержащий адрес медицинской организации или ИП, который:
7.4.1.ОБЯЗАН иметь один [1..1] элемент streetAddressLine, содержащий адрес (текстом), который должен иметь не пустое наполнение;
7.4.2.ОБЯЗАН иметь один [1..1] элемент address:stateCode (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.206", атрибут @codeSystemName должен иметь значение "Субъекты Российской Федерации", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий код субъекта РФ;
7.4.3. ДОЛЖЕН иметь один [1..1] элемент postalCode, содержащий почтовый индекс, который должен иметь не пустое наполнение;
7.4.4.ДОЛЖЕН иметь один [1..1] элемент fias:Address, содержащий идентификатор адреса по ФИАС, который:
7.4.4.1.ОБЯЗАН иметь один [1..1] элемент fias:AOGUID, содержащий глобальный уникальный идентификатор адресного объекта, который должен иметь не пустое наполнение;
7.4.4.2. ДОЛЖЕН иметь один [1..1] элемент fias:HOUSEGUID, содержащий глобальный уникальный идентификатор дома, который должен иметь не пустое наполнение.

**Reply by eugenekogan-spb** [reply]
аналогично организации, указанной в пациенте

**Comment by OlegPenzin** [comment]
итак, есть элемент https://hl7.org/fhir/R4/composition-definitions.html#Composition.author , который может сослаться на ресурс PractitionerRole, который в свою очередь сошлется на Practitioner и Organization

Организацию можно описать отдельно, а вот занимаю должность и самого врача, можно описать по следующим требованиям

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/olegv/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/olegv/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/olegv/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=RU style='tab-interval:35.4pt;word-wrap:break-word'>



Имя | Кард. | Тип | Описание
-- | -- | -- | --
resourceType | [1..1] | code | Тип FHIR-ресурса PractitionerRole
identifier[] | [1..1] | Identifier | Массив идентификатора врача с занимаемой должностью
identifier[0].use | [1..1] | code | Константа – usual
identifier[0].system | [1..1] | uri | Пространство идентификаторов врача в МИС.   Формируется по формуле:   «OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.70»   Например, urn:oid:1.2.643.5.1.13.13.12.2.40.8452.100.1.1.70
identifier[0].value | [1..1] | string | Идентификатор в МИС, например, PRC1324657980
code[] | [1..1] | CodeableConcept | Массив должности врача
code[0].text | [0..1] | string | Наименование должности текстом.   Поле обязательное для заполнения, если не указан код   специальности (code.coding.code)
code[0].coding[0] | [1..1] | Coding | Массив кодирования должности
code[0].coding[0].code | [1..1] | code | Код должности по справочнику НСИ «Должности медицинских и   фармацевтических работников»
code[0].coding[0].system | [1..1] | uri | Идентификатор справочника НСИ «Должности медицинских и   фармацевтических работников»   Константа – urn:oid:1.2.643.5.1.13.13.11.1002
code[0].coding[0].version | [1..1] | string | Номер текущей версии справочника «Должности медицинских и   фармацевтических работников» в НСИ
code[0].coding[0].display | [1..1] | string | Наименование должности по справочнику НСИ «Должности медицинских   и фармацевтических работников»
specialty[] | [0..*] | CodeableConcept | Массив специальности врача
specialty[0].text | [0..1] | string | Наименование специальности текстом.   Поле обязательное для заполнения, если не указан код   специальности (specialty.coding.code)
specialty[0].coding[0] | [1..1] | Coding | Массив кодирования специальности
specialty[0].coding[0].code | [1..1] | code | Код специальности по справочнику НСИ «Номенклатура специальностей   специалистов, имеющих медицинское и фармацевтическое образование»
specialty[0].coding[0].system | [1..1] | uri | Идентификатор справочника НСИ «Номенклатура специальностей   специалистов, имеющих медицинское и фармацевтическое   образование»   Константа – urn:oid:1.2.643.5.1.13.13.11.1066
specialty[0].coding[0].version | [1..1] | string | Номер версии справочника НСИ «Номенклатура специальностей   специалистов, имеющих медицинское и фармацевтическое образование»
specialty[0].coding[0].display | [1..1] | string | Наименование специальности по справочнику НСИ «Номенклатура   специальностей специалистов, имеющих медицинское и фармацевтическое   образование»
organization | [1..1] | Reference | Сведения о медицинской организации, в которой врач занимает   должность
organization.type | [1..1] | uri | Константа – Organization
organization.identifier | [1..1] | Identifier | Идентификатор организации
organization.identifier.system | [1..1] | uri | Константа для ФРМО – urn:oid:1.2.643.5.1.13.13.11.1461
organization.identifier.value | [1..1] | string | Значение OID организации по Реестру медицинских организаций   Российской Федерации, например, 1.2.643.5.1.13.13.12.2.77.8452
practitioner | [1..1] | Reference | Сведения о сотруднике, как физическое лицо
practitioner.type | [1..1] | uri | Константа – Practitioner
practitioner.identifier | [1..1] | Identifier | Идентификатор врача, как физическое лицо
practitioner.identifier.system | [1..1] | uri | Идентификатор СНИЛС врача    Константа – urn:oid:1.2.643.100.3
practitioner.identifier.value | [1..1] | string | СНИЛС врача




</body>

</html>


<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/olegv/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/olegv/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/olegv/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=RU style='tab-interval:35.4pt;word-wrap:break-word'>



Имя | Кард. | Тип | Описание
-- | -- | -- | --
resourceType | [1..1] | code | Тип FHIR-ресурса Practitioner
name[] | [1..1] | HumanName | Массив ФИО врача
name[0].text | [1..1] | string | ФИО врача полностью одной строкой, например Иванов Алексей   Евгеньевич
name[0].family | [1..1] | string | Фамилия врача одной строкой, например, Иванов
name[0].given[] | [1..2] | string | Массив из имени и отчества врача
name[0].given[0] | [1..1] | string | Имя врача одной строкой (первый элемент простого массива   given[]), например, Алексей
name[0].given[1] | [1..1] | string | Отчество врача одной строкой (второй элемент простого массива   given[]), например, Евгеньевич
birthDate | [1..1] | date | Дата рождения с точностью до дня или до года, например,   1989-09-10
identifier[] | [1..*] | Identifier | Массив идентификатора врача
identifier[0].use | [1..1] | code | Константа – official
identifier[0].type.text | [1..1] | string | Константа – СНИЛС
identifier[0].value | [1..1] | string | Номер СНИЛС врача, например, 933 287 402 18
identifier[0].system | [1..1] | uri | Константа – urn:oid:1.2.643.100.3
telecom[] | [0..*] | ContactPoint | Массив контактных данных врача (номер телефона, электронная   почта)
telecom[0].value | [0..1] | string | Номер телефона врача
telecom[0].system | [0..1] | code | Константа – phone.   Значения по справочнику HL7:   phone \| email
telecom[1].value | [1..1] | string | Электронная почта врача
telecom[1].system | [1..1] | code | Константа – email.   Значения по справочнику HL7:   phone \| email \| other
gender | [1..1] | code | Половая принадлежность.   Значения по справочнику HL7:   male \| female \| other \| unknown   Первые 3 значения соответствуют справочнику НСИ Минздрава   России, unknown используется в ситуации отсутствия данных (в качестве аналога   nullFlavor из СЭМД)




</body>

</html>
